### PR TITLE
feat(ci): Support adhoc flamegraph generation + Support `main-cron` flamegraph generation for all queries

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Workflow to trigger adhoc flamegraph generation on arbitrary branch and commit.
You can trigger it via buildkite api, or via buildkite gui after this PR.

---

After #9421 

This will be very useful for https://github.com/risingwavelabs/risingwave/issues/9285.

1. We can `curl` to trigger a build: https://buildkite.com/docs/apis/rest-api/builds#create-a-build.
2. Upload throughput as meta data https://buildkite.com/docs/apis/rest-api/builds#get-a-build.
3. Use that to determine whether tps regressed.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by kwannoel** [comment]
`NEXMARK_QUERIES=all` works:
Screenshot 2023-04-27 at 11 00 35 AM
`NEXMARK_QUERIES="nexmark-q0 nexmark-q101"` works:

Screenshot 2023-04-27 at 11 16 37 AM

**Comment by kwannoel** [comment]
Waiting for https://github.com/risingwavelabs/risingwave/pull/9482 to merge.

**Comment by kwannoel** [comment]
Ready for review

**Review by lmatz — APPROVED** [review]

File: ci/workflows/main-cron.yml [hunk 1]
```diff
@@ -307,4 +307,4 @@ steps:
     # - Nexmark event generation: ~3min for 100mil records.
     # - Generate Flamegraph: ~15min (see https://github.com/koute/not-perf/issues/30 on optimizing)
     # - Building RW artifacts: ~8min
-    timeout_in_minutes: 50
\ No newline at end of file
+    timeout_in_minutes: 120
```
Comment (BugenZhao) [inline ci/workflows/main-cron.yml:310]: Is this enough for all queries? 👀 What about triggering this manually to have a test?

**Review by BugenZhao — APPROVED** [review]
LGTM!

File: ci/workflows/main-cron.yml (hunk 1 above)
Comment (kwannoel) [reply ci/workflows/main-cron.yml:310]: Should be, currently I have one running on `main-cron`: https://buildkite.com/risingwavelabs/main-cron/builds/455#0187c0e5-10c2-42e2-a2b2-86df8484c48a

Comment (kwannoel) [reply ci/workflows/main-cron.yml:310]: Will update on the results

Comment (kwannoel) [reply ci/workflows/main-cron.yml:310]: Sorry I'm wrong, maybe 5 hours is a safer bet.

Comment (kwannoel) [reply ci/workflows/main-cron.yml:310]:
```bash
> ls ../scripts/sql/nexmark/ | grep "q[0-9]*\.sql" | wc -l | sed 's/\(.*\)/\1 \* 15 \/ 60/' | bc
5
```

Assuming worst case of ~15min per query, that would be 5hrs.

Comment (kwannoel) [reply ci/workflows/main-cron.yml]: Trigger a new build: https://buildkite.com/risingwavelabs/main-cron/builds/456

Comment (kwannoel) [reply ci/workflows/main-cron.yml]: Passed in 4hrs, q17 flamegraph looks okay now.

all 23 flamegraphs can be found here: https://buildkite.com/risingwavelabs/cpu-flamegraph/builds/11#0187c1c5-1ac0-4da5-8f6a-4e61a804f832

**Comment by kwannoel** [comment]
^ Minor change to clean up debug log.

**Comment by kwannoel** [comment]
Should also use `jps` to check if `kafka` and `zookeeper` are up. Leave that to another PR. Tracked here: https://github.com/risingwavelabs/risingwave/issues/9426.

**Pull request merged**